lib/configurationSchema.js

```diff
@@ -0,0 +1,438 @@
+        },
+      ],
+    },
+    moduleNameFormatter: {
+      instanceof: 'Function',
+      default: ({ moduleName }) => moduleName,
+    },
+    moduleSideEffectImports: {
+      instanceof: 'Function',
+      default: () => [],
+    },
+    namedExports: {
+      default: {},
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'object',
+          additionalProperties: {
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+        },
+      ],
+    },
+    packageDependencies: {
+      default: ({ config }) =>
+        findPackageDependencies(
+          config.workingDirectory,
+          config.get('importDevDependencies')
+        ),
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'array',
+          items: {
+            type: `string`,
+          },
+        },
+      ],
+    },
+    parserPlugins: {
+      default: DEFAULT_PARSER_PLUGINS,
+      type: 'array',
+      items: {
+        anyOf: [
+          {
+            type: 'string',
+          },
+          {
+            type: 'array',
+            minItems: 2,
+            additionalItems: false,
+            items: [
+              {
+                type: 'string',
+              },
+              {
+                type: 'object',
+                additionalProperties: {
+                  type: 'string',
+                },
+              },
+            ],
+          },
+        ],
+      },
+    },
+    sortImports: {
+      default: true,
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'boolean',
+        },
+      ],
+    },
+    stripFileExtensions: {
+      default: ['.js', '.jsx', '.ts', '.tsx'],
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'array',
+          items: {
+            type: 'string',
+          },
+        },
+      ],
+    },
+    tab: {
+      default: '  ',
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'string',
+        },
+      ],
+    },
+    useRelativePaths: {
+      default: true,
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'boolean',
+        },
+      ],
+    },
+  },
+};
+
+function findGlobalsFromEnvironments(environments) {
+  const result = Object.keys(globals.builtin);
+
+  environments.forEach((environment) => {
+    const envGlobals = globals[environment];
+    if (!envGlobals) {
+      return;
+    }
+    result.push(...Object.keys(envGlobals));
+  });
+  return result;
+}
+
+export function getDefaultConfig() {
+  return Object.entries(SCHEMA.properties).reduce((acc, [k, v]) => {
+    if (typeof v !== `object` || v === null) {
+      throw new Error(`Expected schema key '${k}' to be an object`);
+    }
+    if (v.hasOwnProperty('default')) {
```

Comment:
What do you think about using Object.hasOwn here instead?

Reply:
Either one is okay by me. Possibly one downside of using Object.hasOwn is that we would be losing support for Node <16.8. That might be okay, it's a bit over two years old. I'd probably want a higher benefit/disadvantage ratio.

Maybe we better deal with that the day we choose to use `hasOwnProperty` as a key for our configuration :smile:

Reply:
Yeah, or could use Object.prototype.hasOwnProperty.call